Branch-SDK/src/main/java/io/branch/referral/AttributionReportingManager.kt

```diff
@@ -0,0 +1,166 @@
+            }
+        }
+    }
+
+    fun registerTrigger(context: Context, conversionId: String) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            if (SdkExtensions.getExtensionVersion(SdkExtensions.AD_SERVICES) >= 7) {
+                if (isMeasurementApiEnabled()) {
+                    BranchLogger.v("Registering trigger for conversion ID: $conversionId")
+
+                    val executor = Executors.newSingleThreadExecutor()
+                    val params = getParams(context) + "&conversion_id=$conversionId"
+                    val branchBaseURL = PrefHelper.getInstance(context).apiBaseUrl
+                    val triggerUri = Uri.parse("${branchBaseURL}${Defines.RequestPath.RegisterTrigger}?$params")
+                    val manager = MeasurementManager.get(context)
+
+                    manager.registerTrigger(triggerUri, executor, object : OutcomeReceiver<Any?, Exception> {
+                            override fun onResult(result: Any?) {
+                                BranchLogger.v("Trigger registered successfully with URI: $triggerUri")
+                                executor.shutdown()
+                            }
+
+                            override fun onError(e: Exception) {
+                                BranchLogger.w("Error while registering trigger: ${e.message}")
+                                executor.shutdown()
+                            }
+                        }
+                    )
+                } else {
+                    BranchLogger.v("Measurement API is not enabled. Did not register trigger.")
+                }
+            }
+        }
+    }
+
+    fun getParams(context: Context): String {
+        val systemObserver = DeviceInfo.getInstance().systemObserver
+        val prefHelper = PrefHelper.getInstance(context)
+
+        val gaid = systemObserver.aid
+        val hardwareId = SystemObserver.getUniqueID(context, false).id
+        val brand = SystemObserver.getPhoneBrand()
+        val model = SystemObserver.getPhoneModel()
+        val osVersion = SystemObserver.getOSVersion()
+        val apiLevel = SystemObserver.getAPILevel()
+        val localIp = SystemObserver.getLocalIPAddress()
+        val connectionType = SystemObserver.getConnectionType(context)
+        val carrier = SystemObserver.getCarrier(context)
+        val screenDisplay = SystemObserver.getScreenDisplay(context)
+        val dpi = screenDisplay.densityDpi
+        val screenWidth = screenDisplay.widthPixels
+        val screenHeight = screenDisplay.heightPixels
+        val uiMode = SystemObserver.getUIMode(context)
+        val osName = SystemObserver.getOS(context)
+        val language = SystemObserver.getISO2LanguageCode()
+        val country = SystemObserver.getISO2CountryCode()
+        val sandboxVersion = when {
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.R &&
+                    SdkExtensions.getExtensionVersion(Build.VERSION_CODES.R) >= 4 -> {
```

Comment:
Where does this value come from?

Reply:
Im realizing that this version check was no longer necessary because of the earlier ones I added to the actual method that call getParams